web/src/components/Upstream/UpstreamForm.tsx

```diff
@@ -86,14 +87,18 @@ const UpstreamForm: React.FC<Props> = forwardRef(
         if (required) {
           requestAnimationFrame(() => {
             form.resetFields();
-            form.setFieldsValue(DEFAULT_UPSTREAM);
             setHiddenForm(false);
           });
         }
       } else {
         if (upstream_id) {
           requestAnimationFrame(() => {
-            form.setFieldsValue(list.find((item) => item.id === upstream_id));
+            const targetData = list.find((item) => item.id === upstream_id) as UpstreamComponent.ResponseData
+            if (targetData) {
+              form.setFieldsValue(transformUpstreamDataFromRequest(targetData));
+            } else {
+              // TODO: 提示 upstream_id 找不到想要的数据
```

Comment:
@juzhiyuan should we use English here? Thanks.

Reply:
sure! I missed some notes, and will file a new PR to remove all Chinese.

Reply:
yes, English only

Reply:
I just filed a PR to make the Upstream Logics more clear https://github.com/apache/apisix-dashboard/pull/1784, no need to check if target data lost any more.